modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -192,3 +196,29 @@
                    (map
                     (partial into {})
                     (db/select [Field :name :database_type :base_type] :table_id table-id))))))))))
+
+(defn- assert-jdbc-url-fetch-size [db fetch-size]
+  (let [ds   (sql-jdbc.execute/datasource db)
+        conn (.getConnection ds)
+        md   (.getMetaData conn)
+        url  (.getURL md)]
+    (is (str/includes? url (str "defaultRowFetchSize=" fetch-size)))))
+
+(deftest test-jdbc-fetch-size
+  (testing "Redshift JDBC fetch size is set correctly in PreparedStatement"
+    (mt/test-driver :redshift
+      (assert-jdbc-url-fetch-size (mt/db) 5000) ; the default value should always be picked up if nothing is set
```

Comment:
Generally it's better to avoid magic numbers like this from a code readability standpoint. It's also one less line that has to change if we change the default value.

You can do something like

```clj
(:default (setting/resolve-setting :redshift-fetch-size))
```

to get the default value

Reply:
Done.